packages/relay-kit/src/packs/safe-4337/Safe4337Pack.ts

```diff
@@ -415,17 +419,67 @@ export class Safe4337Pack extends RelayKitBasePack<{
     })
   }
 
+  /**
+   * Converts a SafeOperationResponse to an EthSafeOperation.
+   *
+   * @param {SafeOperationResponse} safeOperationResponse - The SafeOperationResponse to convert to EthSafeOperation
+   * @returns {EthSafeOperation} - The EthSafeOperation object
+   */
+  #toSafeOperation(safeOperationResponse: SafeOperationResponse): EthSafeOperation {
+    const { validUntil, validAfter, userOperation } = safeOperationResponse
+
+    const safeOperation = new EthSafeOperation(
+      {
+        sender: userOperation?.sender || '0x',
+        nonce: userOperation?.nonce?.toString() || '0',
+        initCode: userOperation?.initCode || '',
+        callData: userOperation?.callData || '',
+        callGasLimit: BigInt(userOperation?.callGasLimit || 0n),
+        verificationGasLimit: BigInt(userOperation?.verificationGasLimit || 0),
+        preVerificationGas: BigInt(userOperation?.preVerificationGas || 0),
+        maxFeePerGas: BigInt(userOperation?.maxFeePerGas || 0),
+        maxPriorityFeePerGas: BigInt(userOperation?.maxPriorityFeePerGas || 0),
+        paymasterAndData: userOperation?.paymasterData || '0x',
+        signature: userOperation?.signature || '0x'
+      },
+      {
+        moduleAddress: this.#SAFE_4337_MODULE_ADDRESS,
+        entryPoint: userOperation?.entryPoint || this.#ENTRYPOINT_ADDRESS,
+        validAfter: validAfter ? new Date(validAfter).getTime() : undefined,
+        validUntil: validUntil ? new Date(validUntil).getTime() : undefined
+      }
+    )
+
+    if (safeOperationResponse.confirmations) {
+      safeOperationResponse.confirmations.forEach((confirmation: SafeOperationConfirmation) => {
+        safeOperation.addSignature(new EthSafeSignature(confirmation.owner, confirmation.signature))
+      })
+    }
+
+    return safeOperation
+  }
+
```

Comment:
Don't you feel we are adding lot of utils like functionality into the main class? We even have some other utils at the bottom of this file that makes the file to be very long.

Reply:
yes, you are right